File: python/06.py
from common import *

with open(infile('06.txt'), 'r') as f:
    data = [0 for i in range(11)]
    for v in get_all_ints(f.read()):
        data[v] += 1

def solve(iters):
    vals = data.copy()
    for i in range(iters):
        vals[9] += vals[0]
        vals[7] += vals[0]
        vals[:-1] = vals[1:]
    return sum(vals)

print("Part 1:", solve(80))
print("Part 1:", solve(256))

File: c/06.c
#include "common.h"

u64 counts[11] = {0};

void parse() {
  FILE *f = fopen(INFILE("06.txt"), "r");
  char buf[2048];
  fgets(buf, 2048, f);

  memset(counts, 0, sizeof(counts));
  char *line = buf, *tmp;
  while ((tmp = strsep(&line, ",")))
    counts[atoi(tmp)]++;

  fclose(f);
}

u64 solve(int iters) {
  parse();
  for (int i = 0; i < iters; i++) {
    counts[7] += counts[0];
    counts[9] += counts[0];
    memcpy(counts, counts + 1, 10 * sizeof(counts[0]));
  }
  u64 result = 0;
  for (int i = 0; i < 12; i++)
    result += counts[i];
  return result;
}

int main() {
  printf("Part 1: %llu\n", solve(80));
  printf("Part 2: %llu\n", solve(256));
  return 0;
}